packages/@ember/-internals/metal/lib/chain-tags.ts

```diff
@@ -104,20 +105,28 @@ export function getChainTagsForKey(obj: any, path: string) {
     if (segment === 'args' && ARGS_PROXY_TAGS.has(current.args)) {
       assert(
         `When watching the 'args' on a GlimmerComponent, you must watch a value on the args. You cannot watch the object itself, as it never changes.`,
-        segmentEnd !== path.length
+        segmentEnd !== pathLength
       );
 
-      segment = path.substr(segmentEnd + 1)!;
-      segmentEnd = path.indexOf('.', segmentEnd);
+      lastSegmentEnd = segmentEnd + 1;
+      segmentEnd = path.indexOf('.', lastSegmentEnd);
+
+      if (segmentEnd === -1) {
+        segmentEnd = pathLength;
+      }
+
+      segment = path.slice(lastSegmentEnd, segmentEnd)!;
 
       let namedArgs = ARGS_PROXY_TAGS.get(current.args);
       let ref = namedArgs.get(segment);
 
       chainTags.push(ref.tag);
 
-      if (segmentEnd !== -1) {
+      if (segmentEnd !== pathLength) {
         current = ref.value();
-        segment = path.substr(segmentEnd + 1);
+        // lastSegmentEnd = segmentEnd + 1;
```

Comment:
Hmm, do we need these comments? Should we remove them?

Reply:
ah, that was a mistake, definitely